pinc/Project.inc

```diff
@@ -686,6 +686,31 @@ XML;
         return $this->state == PROJ_POST_FIRST_CHECKED_OUT &&
                $this->smoothread_deadline > time();
     }
+
+    private function delete_file($file)
+    {
+        if(is_dir($file))
```

Comment:
Let's add a check to ensure that `$file` is always inside `$this->dir` and throw an UnexpectedValueException if it isn't. Maybe something like (totally untested):
```php
// ensure $file is inside $this->dir
if(! startswith(realpath($file), realpath($this->dir)))
{
    throw new UnexpectedValueException("$file not in $this->dir");
}
```
realpath ensures that no funny path business is going on that might do something outside the directory we're expecting.